packages/wonder-stuff-sentry/src/__tests__/sentry-data-reducer.test.js

```diff
@@ -0,0 +1,186 @@
+            fingerprint: [],
+        };
+
+        // Act
+        const result = sentryDataReducer(accumulator, current, 0, [current]);
+
+        // Assert
+        expect(result).toEqual(accumulator);
+    });
+
+    it("should return object matching current if accumulator is empty", () => {
+        // Arrange
+        const accumulator: SentryData = {
+            tags: {},
+            contexts: {},
+            fingerprint: [],
+        };
+        const current = {
+            tags: {
+                tag1: "value1",
+            },
+            contexts: {
+                context1: {
+                    value1: "value1value",
+                },
+            },
+            fingerprint: ["a"],
+        };
+
+        // Act
+        const result = sentryDataReducer(accumulator, current, 0, [current]);
+
+        // Assert
+        expect(result).toEqual(current);
+    });
+
+    it("should return combine and deduplicate fingerprint values between accumulator and current", () => {
+        // Arrange
+        const accumulator: SentryData = {
+            tags: {},
+            contexts: {},
+            fingerprint: ["a", "c"],
+        };
+        const current = {
+            tags: {},
+            contexts: {},
+            fingerprint: ["a", "b"],
+        };
+
+        // Act
+        const result = sentryDataReducer(accumulator, current, 0, [current]);
+
+        // Assert
+        expect(result).toEqual({
+            tags: {},
+            contexts: {},
+            fingerprint: ["a", "c", "b"],
+        });
+    });
+
+    it("should return a combination of accumulator and current tags and contexts, with current winning collisions", () => {
+        // Arrange
+        const accumulator: SentryData = {
+            tags: {
+                tag1: "value1Accumulator",
+                tag2: "value2Accumulator",
+            },
+            contexts: {
+                context1: {
+                    value1: "value1Accumulator",
+                },
+                context2: {
+                    value2: "value2Accumulator",
+                },
+            },
+            fingerprint: [],
+        };
+        const current = {
+            tags: {
+                tag1: "value1Current",
+                tag3: "value3Current",
+            },
+            contexts: {
+                context1: {
+                    value1: "value1Current",
+                },
+                context3: {
+                    value3: "value3Current",
+                },
+            },
+            fingerprint: [],
+        };
+
+        // Act
+        const result = sentryDataReducer(accumulator, current, 0, [current]);
+
+        // Assert
+        expect(result).toEqual({
+            tags: {
+                tag1: "value1Current",
+                tag2: "value2Accumulator",
+                tag3: "value3Current",
+            },
+            contexts: {
+                context1: {
+                    value1: "value1Current",
```

Comment:
The values you picked make it really easy to see what's going on in these tests.

Reply:
Agreed!!

Maybe worth checking all the unit tests in the mobile code base to make sure lots of combos are covered?